Documentation/Books/stash/DocumentRevisionMVCC.md

```diff
@@ -0,0 +1,57 @@
+As ArangoDB supports MVCC (Multiple Version Concurrency Control),
+documents can internally exist in more than one revision.
+The document revision is the MVCC token used to specify
+a particular revision of a document (identified by its `_id`).
+
+
+
+It is a string value that contained (up to ArangoDB 3.0)
+an integer number and is unique within the list of document
+revisions for a single document. 
+In ArangoDB >= 3.1 the _rev strings
+are in fact time stamps. They use the local clock of the DBserver that
+actually writes the document and have millisecond accuracy. 
+A [_Hybrid Logical Clock_](http://www.cse.buffalo.edu/tech-reports/2014-04.pdf)
+is used.
+
+Within one shard it is guaranteed that two different document revisions
+have a different `_rev` string, even if they are written in the same
+millisecond, and that these stamps are ascending.
+
+Note however that different servers in your cluster might have a clock
+skew, and therefore between different shards or even between different
+collections the time stamps are not guaranteed to be comparable.
```

Comment:
In a single server setup, `_rev` values are guaranteed to be unique across all documents and all collections.